developing_packages/editable_packages.rst

```diff
@@ -0,0 +1,245 @@
++++++++++++++++++++++++
+
+Layout files are specified in the :command:`conan link` command, as an extra argument:
+
+.. code-block:: bash
+
+    $ conan link . cool/version@user/dev --layout=win_layout
+
+That ``win_layout`` file will be first looked for relative to the current directory (the
+path can be absolute too). If it is found, that will be used. It is possible to add those
+layouts in the source repositories, so they are always easy to find after a clone.
+
+If the specified layout is not found relative to the current directory, it will be looked
+for in the conan cache, in the ``.conan/layouts`` folder. This is very convenient to have
+a single definition of layouts that can be shared with the team and installed with
+``conan config install``.
+
+If no argument is specified, the :command:`conan link` command will try to use a `.conan/layouts/default`
+layout from the local cache.
+
+You can switch layout files by passing a different argument to new calls to :command:`conan link`.
+
+Evaluation order and priority
++++++++++++++++++++++++++++++
+
+It is important to understand the evaluation order and priorities regarding the definitions of layouts:
+
+- The first thing that will always execute is the recipe ``package_info()``. That will define
+  the flags, definitions, as well as some values for the layout folders: ``includedirs``, ``libdirs``, etc.
+- If a layout file is defined, either explicitly or using the implicit ``.conan/layouts/default``,
+  conan will look for matches, based on its package reference.
+- If a match is found, either because of global definitions like ``[includedirs]``
+  or because a match like ``[pkg/version@user/channel:includedirs]``, then the layout folders
+  (includedirs, libdirs, resdirs, builddirs, bindirs), will be invalidated and replaced by the ones
+  defined in the file.
+- If a specific match like ``[pkg/version@user/channel:includedirs]`` is found, it is expected to
+  have defined also its specific ``[pkg/version@user/channel:libdirs]``, etc. The global layout
+  folders specified without package reference won't be applied once a match is found.
+- It no match is found, the original values for the layout folders defined in ``package_info()`` will
+  be respected.
+- The layout file to be used is defined at ``conan link`` time. If a ``.conan/layouts/default`` file
+  is added after the ``conan link``, it will not be used at all.
+
+
+Using a package in editable mode
+--------------------------------
+
+Once a reference is in editable mode it is used **system wide** (for every set of ``settings`` and
+``options``) by Conan (by every Conan client that uses the same cache), no changes are
+required in the consumers. Every :command:`conan install` command that requires our editable
+``cool/version@user/dev`` package will use the paths to the local directory and the changes
+made to this project will be taken into account by the packages using its headers or linking
+against it. 
```

Comment:
I think you previously mentioned that some commands will be unavailable for editable packages (e.g. `conan upload` or `conan export`), right?
if so, may be we can clearly document which commands aren't supported for editable packages here?